app/lib/katello/resources/registry.rb

```diff
@@ -26,18 +26,12 @@ def load_class
             registry_url = nil
             pulp_primary = ::SmartProxy.pulp_primary
 
-            # Pulp 3 has its own registry
-            if pulp_primary&.pulp3_repository_type_support?(::Katello::Repository::DOCKER_TYPE)
-              uri = URI(pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url'))
-              uri.path = "/pulpcore_registry/"
-              registry_url = uri.to_s
-
-              # Assume the registry uses the same CA as the Smart Proxy
-              ca_cert_file = Setting[:ssl_ca_file]
-            elsif (container_config = SETTINGS.dig(:katello, :container_image_registry))
-              registry_url = container_config[:crane_url]
-              ca_cert_file = container_config[:registry_ca_cert_file]
-            end
+            uri = URI(pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url'))
+            uri.path = "/pulpcore_registry/"
+            registry_url = uri.to_s
+
+            # Assume the registry uses the same CA as the Smart Proxy
+            ca_cert_file = Setting[:ssl_ca_file]
 
             fail Errors::ContainerRegistryNotConfigured unless registry_url
```

Comment:
Is this still reachable code? I don't think it is. I think it should fail `unless pulp_primary.setting(SmartProxy::PULP3_FEATURE, 'content_app_url')` is set.